src/sonic-py-common/sonic_py_common/device_info.py

```diff
@@ -600,19 +600,21 @@ def get_system_mac(namespace=None):
         machine_vars = get_machine_info()
         (mac, err) = run_command(syseeprom_cmd)
         hw_mac_entry_outputs.append((mac, err))
-        if not mac:
-            if machine_vars is not None and machine_key in machine_vars:
-                hwsku = machine_vars[machine_key]
-                profile_cmd0 = ['cat', HOST_DEVICE_PATH + '/' + platform + '/' + hwsku + '/profile.ini']
+        if machine_vars is not None and machine_key in machine_vars:
+            hwsku = machine_vars[machine_key]
+            profile_file = HOST_DEVICE_PATH + '/' + platform + '/' + hwsku + '/profile.ini'
+            if os.path.exists(profile_file):
+                profile_cmd0 = ['cat', profile_file]
                 profile_cmd1 = ['grep', 'switchMacAddress']
                 profile_cmd2 = ['cut', '-f2', '-d', '=']
                 (mac, err) = run_command_pipe(profile_cmd0, profile_cmd1, profile_cmd2)
-            else:
-                profile_cmd = ["false"]
-                (mac, err) = run_command(profile_cmd)
-            hw_mac_entry_outputs.append((mac, err))
-            (mac, err) = run_command_pipe(iplink_cmd0, iplink_cmd1, iplink_cmd2)
+                hw_mac_entry_outputs.append((mac, err))
+        else:
+            profile_cmd = ["false"]
+            (mac, err) = run_command(profile_cmd)
             hw_mac_entry_outputs.append((mac, err))
+        (mac, err) = run_command_pipe(iplink_cmd0, iplink_cmd1, iplink_cmd2)
```

Comment:
@pavannaregundi line 616 overwrites the value of (mac, err) obtained in line 601. So why do we have line 601. What about Marvell platform that may return mac, error correctly at line 601

Reply:
@prgeor No. hw_mac_entry_outputs is a list has the appended value of (mac,err) from 601. So value is not lost. Logic is try get mac from syseeprom and hwsku(profile.ini) and eth0 in same order and return first valid mac from this list.